src/quantuminspire/version.py

```diff
@@ -1 +1 @@
-__version__ = '2.0.0'
+__version__ = '2.1.0'
```

Comment:
Some of the changes in this PR appear to be breaking changes. Specifically, I notice that the default value of `full_state_projection` was switched in some places, and the dependency changes are also breaking since we dropped support for Python 3.7. By SemVer 2.0, shouldn't the major version be increased?

Reply:
We didn't drop support for python 3.7 (see setup.py). We only dropped the travis build for python 3.7 (because dependencies problems on their environment).